nessie/src/test/java/org/apache/iceberg/nessie/TestBranchVisibility.java

```diff
@@ -455,7 +456,9 @@ public void testWithRefAndHash() throws NessieConflictException, NessieNotFoundE
     String hashBeforeNamespaceCreation = api.getReference().refName(testBranch).get().getHash();
     Namespace namespaceA = Namespace.of("a");
     Namespace namespaceAB = Namespace.of("a", "b");
-    assertThat(nessieCatalog.listNamespaces(namespaceAB)).isEmpty();
+    NessieCatalog finalNessieCatalog = nessieCatalog;
+    assertThatThrownBy(() -> finalNessieCatalog.listNamespaces(namespaceAB))
+        .isInstanceOf(NoSuchNamespaceException.class);
```

Comment:
this should also have a `.hasMessage()` check. Same for the other places in this file